instance/models/mixins/openedx_database.py

```diff
@@ -75,23 +75,17 @@ def mongo_database_names(self):
         """
         return [self.mongo_database_name, self.forum_database_name]
 
-    def provision_mysql(self):
+    def set_field_defaults(self):
         """
-        Set mysql credentials and provision the database.
+        Set default values for mysql and mongo credentials.
         """
-        if not self.mysql_provisioned:
-            self.mysql_user = get_random_string(length=16, allowed_chars=string.ascii_lowercase)
-            self.mysql_pass = get_random_string(length=32)
-        return super().provision_mysql()
-
-    def provision_mongo(self):
-        """
-        Set mongo credentials and provision the database.
-        """
-        if not self.mongo_provisioned:
-            self.mongo_user = get_random_string(length=16, allowed_chars=string.ascii_lowercase)
-            self.mongo_pass = get_random_string(length=32)
-        return super().provision_mongo()
+        if not self.use_ephemeral_databases:
+            if not self.mysql_provisioned:
```

Comment:
Actually, I realized there is a problem with making this into `set_field_defaults`: If I create an instance with `use_ephemeral_databases=True` but then edit it so it has `use_ephemeral_databases=False` (which is totally legitimate), then `spawn_appserver()` will fail because `mysql_user` and `mysql_pass` will never be generated.

One fix for this could be to always generate `mysql_user` regardless of whether `set_field_defaults` is set. Since we want to eventually drop support for ephemeral databases entirely, I think that is reasonable - as long as it won't interfere with `edxlocal` creating local DBs in the ephemeral case.